core/src/main/java/nl/nn/adapterframework/core/Adapter.java

```diff
@@ -689,8 +689,8 @@ public PipeLineResult processMessageWithExceptions(String messageId, Message mes
 			if (log.isDebugEnabled()) {
 				log.debug("Adapter: [{}] STAT: Pipeline finished processing message with messageId [{}] exit-state [{}] started {} finished {} total duration: {} ms",
 						getName(), messageId, result.getState(),
-						DateFormatUtils.format(new Date(startTime), DateFormatUtils.FORMAT_FULL_GENERIC),
-						DateFormatUtils.format(new Date(endTime), DateFormatUtils.FORMAT_FULL_GENERIC),
+						DateFormatUtils.format(startTime, DateFormatUtils.FORMAT_FULL_GENERIC),
```

Comment:
Instead of passing the String FORMATs everywhere you can also pass perhaps the constants for the FORMATTERS? Especially when you move to a new java.time API, since these new classes should be thread-safe and therefore safe to re-use (which the old classes were not).

This would save a lot of object-creation with every date formatted and thus reap the benefits of using the new API.

PS: This comment applies to the whole code-base of course not just this single line but I don't want to repeat it throughout the whole PR!